networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandler.java

```diff
@@ -73,14 +81,19 @@ public void onIncomingMessage(
     LOG.trace("Peer {} requested BeaconBlocks with roots: {}", peer.getId(), message);
     if (storageClient.getStore() != null) {
       SafeFuture<Void> future = SafeFuture.COMPLETE;
-      if (!peer.wantToMakeRequest() || !peer.wantToReceiveBlocks(callback, message.size())) {
+      if (!peer.wantToMakeRequest()
+          || !peer.wantToReceiveBlocks(
+              callback, maxRequestSize.min(UInt64.valueOf(message.size())).longValue())) {
         requestCounter.labels("rate_limited").inc();
         return;
       }
-
       requestCounter.labels("ok").inc();
       totalBlocksRequestedCounter.inc(message.size());
-      for (SszBytes32 blockRoot : message) {
+
+      final int count = Math.min(maxRequestSize.intValue(), message.size());
```

Comment:
we can do it before `if (!peer.wantToMakeRequest()` and calculate it only once if I'm not wrong

Reply:
Done